### PR TITLE
Convert some previously ignored integration tests to unit tests

### LINKED ISSUE #1694: Decide what to do about the #[ignore]d tests due to bootstrap blacklist missing in quic-p2p
quic-p2p doesn't support bootstrap blacklist yet, so some tests that relied on it were `#[ignore]`d. Decide what to do about them: either implement bootstrap blacklist in quic-p2p and then remove the `#[ignore]` attribute, or modify the tests to not require bootstrap blacklist (perhaps by converting them to unit tests - similar to the tests in `src/states/elder/tests.rs`) or remove them.

### PR DESCRIPTION
**Pull request by madadam**
Closes #1694

### CODE REVIEW

File: src/peer_manager.rs [hunk 1]
```diff
@@ -511,27 +511,13 @@ impl PeerManager {
...
-
-        // Allow only one client per IP
-        self.peers
-            .values()
-            .filter_map(Peer::client_ip)
-            .filter(|other_ip| *other_ip == client_ip)
-            .take(2)
-            .count()
-            >= 2
+    /// Checks whether we can accept more clients.
```
Comment (pierrechevalier83) [inline src/peer_manager.rs]: I would add a line with "Allow one client per IP". Personally, it helped me understand the implementation.

File: src/states/elder/tests.rs [hunk 2]
```diff
@@ -52,6 +54,30 @@ impl CandidateInfo {
     }
 }
 
+struct ClientInfo {
+    full_id: FullId,
+    addr: SocketAddr,
+}
+
+impl ClientInfo {
+    fn with_addr(addr: &str) -> Self {
```
Comment (maqi) [inline src/states/elder/tests.rs]: normally just `new` for the constructor?
`with_addr` indicating there is another constructor `without addr` ?

File: src/states/elder/tests.rs [hunk 3]
```diff
@@ -64,6 +90,10 @@ struct ElderUnderTest {
 
 impl ElderUnderTest {
     fn new() -> Self {
+        Self::with_min_section_size(DEFAULT_MIN_SECTION_SIZE)
```
Comment (maqi) [inline src/states/elder/tests.rs]: hmm... so there is other places calling this constructor with different size?

File: src/states/elder/tests.rs [hunk 4]
```diff
@@ -413,7 +469,7 @@ fn new_elder_state(
     let parsec_map = ParsecMap::new(full_id.clone(), gen_pfx_info);
     let chain = Chain::new(min_section_size, public_id, gen_pfx_info.clone());
     let peer_map = PeerMap::new();
-    let peer_mgr = PeerManager::new(public_id, true);
+    let peer_mgr = PeerManager::new(public_id, false);
```
Comment (maqi) [inline src/states/elder/tests.rs]: why the flag needs to be changed from `true` to `false`?

File: tests/mock_network/requests.rs [hunk 5]
```diff
@@ -227,77 +225,3 @@ fn failed_get_request() {
...
     assert_eq!(response_received_count, 1);
 }
-
-#[test]
-// TODO (quic-p2p): this test relies on a behaviour of mock-crust which drops all in-flight messages
-// on disconnect. Verify whether this behaviour should be transitioned over to quic-p2p. If yes,
-// then do so and then re-enable this test. If not, modify this test accordignly (or remove it if
-// not relevant anymore).
-#[ignore]
-fn disconnect_on_get_request() {
```
Comment (maqi) [inline tests/mock_network/requests.rs]: the title of the PR indicates this ignored test shall be moved to somewhere else.
however it seems not appear among the above new created tests?

File: src/states/elder/tests.rs (hunk 2 above)
Comment (pierrechevalier83) [reply src/states/elder/tests.rs]: I think `new` is idiomatic for a constructor that takes no argument. If you take an argument, especially a `&str` where the type isn't particularly meaningful, `with_addr` is a better name as it tells the user what argument they should pass with just the signature. The name of the variable is local and not part of the signature.

File: tests/mock_network/requests.rs (hunk 5 above)
Comment (pierrechevalier83) [reply tests/mock_network/requests.rs]: If you look at it commit by commit, some are moved and some are removed. When tests are removed, Adam explained why in the appropriate commit messages.

File: src/states/elder/tests.rs (hunk 2 above)
Comment (madadam) [reply src/states/elder/tests.rs]: I think I originally wanted to also nave `new` which would return a default or random address, but ended up not needing it. I'm fine with renaming it to `new`, but I think @pierrechevalier83 has a valid point.

File: src/states/elder/tests.rs (hunk 3 above)
Comment (madadam) [reply src/states/elder/tests.rs]: Yes, in the `accept_previously_rejected_client_after_reaching_min_section_size` test.

File: src/states/elder/tests.rs (hunk 4 above)
Comment (madadam) [reply src/states/elder/tests.rs]: This flag control whether client IP limiting is enabled (`true` = disable, `false` = not disable). I needed it to be enabled for the new tests to work. It doesn't affect the tests that were already there.

**Pull request merged**